src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1583,7 +1634,7 @@ def set_up_pod_identity_profile(self, mc: ManagedCluster) -> ManagedCluster:
         enable_pod_identity = self.context.get_enable_pod_identity()
         enable_pod_identity_with_kubenet = self.context.get_enable_pod_identity_with_kubenet()
         if enable_pod_identity:
-            pod_identity_profile = self.models.ManagedClusterPodIdentityProfile(
+            pod_identity_profile = self.models.pod_identity_models.get("ManagedClusterPodIdentityProfile")(
```

Comment:
is it type safe? This looks easy to have typo here

Reply:
This should be safe. `self.models` refers to an instance of class `AKSPreviewModels`, which is a holder of the model definitions. This class will help load all the models of a certain api version through the `resource_type` parameter, which avoids hard-coding the import model of a certain api version in each file.

Further for this case, we store the two models of `ManagedClusterPodIdentityProfile` and `ManagedClusterPodIdentityException` in the `pod_identity_models` property of this class. This attribute is a dictionary and will be set within the initialization process of this class.